scripts/download_checkpoints.py

```diff
@@ -0,0 +1,137 @@
+    save_path.parent.mkdir(parents=True, exist_ok=True)
+    
+    with open(save_path, 'wb') as f:
+        with tqdm(total=total_size, unit='B', unit_scale=True, desc=save_path.name) as pbar:
+            for chunk in response.iter_content(chunk_size=chunk_size):
+                if chunk:
+                    f.write(chunk)
+                    pbar.update(len(chunk))
+
+def try_get_directory_listing(url):
+    common_files = [
+        "config.yaml",
+        "model.pt",
+        "optim.pt",
+        "train.pt",
+        "model.safetensors",
+        "optim.safetensors",
+    ]
+    found_files = []
+    for pattern in common_files:
+        try:
+            test_url = urljoin(url.rstrip('/') + '/', pattern)
+            response = requests.head(test_url)
+            # response.raise_for_status()
+            if response.status_code == 200:
+                found_files.append(pattern)
+        except requests.exceptions.HTTPError as e:
+            print(f"HTTP error for {pattern}: {e}")
+        except requests.exceptions.RequestException as e:
+            print(f"Connection error for {pattern}: {e}")
+    return found_files
+
+def download_checkpoint(url, save_dir):
+   """Download all files from a checkpoint directory."""
+   r2_url = convert_to_r2_url(url)
+   public_url = convert_to_public_url(r2_url)
+   base_path = Path(save_dir)
+   base_path.mkdir(parents=True, exist_ok=True)
+   print(f"Saving to: {base_path}")
+   available_files = try_get_directory_listing(public_url)
+   
+   if not available_files:
+       raise ValueError("No matching files found in directory")
+   
+   failed_files = []
+   for file in available_files:
+       file_url = urljoin(public_url.rstrip('/') + '/', file)
+       file_path = base_path / file
+       try:
+           print(f"\nDownloading: {file}")
+           download_file(file_url, file_path)
+       except requests.exceptions.Timeout:
+           print(f"Timeout error for {file}, retrying once...")
+           try:
+               download_file(file_url, file_path)
+           except requests.exceptions.RequestException as e:
+               failed_files.append(file)
+               print(f"Failed to download {file}: {e}")
+       except requests.exceptions.RequestException as e:
+           failed_files.append(file)
+           print(f"Failed to download {file}: {e}")
+   if failed_files:
+       print(f"\nWARNING: Failed to download these files: {failed_files}")
+
+def main():
+    parser = argparse.ArgumentParser(description='Download OLMo checkpoints')
+    subparsers = parser.add_subparsers(dest='command', help='Commands')
+    download_parser = subparsers.add_parser('download', 
+                                          help='Download checkpoints from CSV file')
+    download_parser.add_argument('csv_file', type=str, 
+                               help='Path to the CSV file containing checkpoint URLs')
+    download_parser.add_argument('--step', type=str, required=True,
```

Comment:
In the code it seems like `step` is not required if you want to download everything?

Reply:
No, step is required argument. It throws this error when user fails to add `--step`.
```bash
python scripts/download_checkpoints.py download checkpoints/official/OLMo-1B.csv  --save-dir ./new_checkpoints 
usage: download_checkpoints.py download [-h] --step STEP [--save-dir SAVE_DIR] csv_file
download_checkpoints.py download: error: the following arguments are required: --step
```